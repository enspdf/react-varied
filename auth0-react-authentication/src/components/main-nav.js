import { NavLink } from "react-router-dom";
import { useAuth0 } from "@auth0/auth0-react";

const MainNav = () => {
  const { loginWithRedirect, logout, user, isLoading } = useAuth0();

  return (
    <div className="navbar-nav mr-auto">
      <NavLink
        to="/"
        exact
        className="nav-link"
        activeClassName="router-link-exact-active"
      >
        Home
      </NavLink>
      {user && (
        <NavLink
          to="/profile"
          exact
          className="nav-link"
          activeClassName="router-link-exact-active"
        >
          Profile
        </NavLink>
      )}
      <NavLink
        to="/external-api"
        exact
        className="nav-link"
        activeClassName="router-link-exact-active"
      >
        External API
      </NavLink>
      {!isLoading && !user && (
        <button
          className="btn btn-primary btn-block"
          onClick={loginWithRedirect}
        >
          Log In
        </button>
      )}
      {!isLoading && user && (
        <button className="btn btn-primary btn-block" onClick={logout}>
          Log Out
        </button>
      )}
    </div>
  );
};

export default MainNav;
