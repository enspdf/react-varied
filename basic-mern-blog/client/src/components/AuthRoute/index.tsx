import React, { useContext } from 'react'
import { Redirect } from 'react-router'
import logging from '../../config/logging'
import UserContext from '../../context/user'

export interface IAuthRouteProps {
    children?: any
}

const AuthRoute = ({ children }: IAuthRouteProps) => {
    const { user } = useContext(UserContext).userState

    if (user._id === '') {
        logging.info('Unathorized, redirecting ...')

        return <Redirect to="/login" />
    } else {
        return <>{children}</>
    }
}

export default AuthRoute
