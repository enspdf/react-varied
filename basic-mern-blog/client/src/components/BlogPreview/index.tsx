import React from 'react'
import { Link } from 'react-router-dom'
import { Card, CardBody } from 'reactstrap'

export interface IBlogPreviewProps {
    children?: any
    _id: string
    title: string
    headline: string
    author: string
    createdAt: string
    updatedAt: string
}

const BlogPreview = ({ children, _id, title, headline, author, createdAt, updatedAt }: IBlogPreviewProps) => {
    return (
        <Card className="border-0">
            <CardBody className="p-0">
                <Link to={`/blogs/${_id}`} style={{ textDecoration: 'none' }} className="text-primary">
                    <h1>
                        <strong>{title}</strong>
                    </h1>
                    <h3>{headline}</h3>
                    <br />
                </Link>
                {createdAt !== updatedAt ? (
                    <p>
                        Updated by {author} at {new Date(updatedAt).toLocaleDateString()}
                    </p>
                ) : (
                    <p>
                        Posted by {author} at {new Date(createdAt).toLocaleDateString()}
                    </p>
                )}
                {children}
            </CardBody>
        </Card>
    )
}

export default BlogPreview
