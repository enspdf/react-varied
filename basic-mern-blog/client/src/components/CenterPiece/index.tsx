import React from 'react'
import { Container } from 'reactstrap'

export interface ICenterPieceProps {
    children: any
}

const CenterPiece = ({ children }: ICenterPieceProps) => {
    return (
        <Container fluid className="p-0">
            <Container style={{ position: 'absolute', left: '50%', top: '50%', transform: 'translate(-50%, -50%)' }} className="d-flex justify-content-center">
                {children}
            </Container>
        </Container>
    )
}

export default CenterPiece
