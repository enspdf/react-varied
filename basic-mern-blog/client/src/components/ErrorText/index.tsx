import React from 'react'

export interface IErrorTextProps {
    error: string
}

const ErrorText = ({ error }: IErrorTextProps) => {
    if (error === '') return null

    return <small className="text-danger">{error}</small>
}

export default ErrorText
