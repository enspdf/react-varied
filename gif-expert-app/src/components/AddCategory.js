import PropTypes from "prop-types";
import React, { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
