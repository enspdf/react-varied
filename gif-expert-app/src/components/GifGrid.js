import React, { useEffect, useState } from "react";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h1 className="animate__animated animate__fadeIn">{category}</h1>
      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
