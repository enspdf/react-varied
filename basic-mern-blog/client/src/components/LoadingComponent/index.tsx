import React from 'react'
import { Card, CardBody } from 'reactstrap'
import CenterPiece from '../CenterPiece'

export interface ILoadingProps {
    children: any
    dotType?: string
}

export const Loading = ({ children, dotType }: ILoadingProps) => {
    return (
        <div className="text-center">
            <div className="stage">
                <div className={dotType} />
            </div>
            {children}
        </div>
    )
}

Loading.defaultProps = {
    dotType: 'dot-bricks'
}

export interface ILoadingComponentProps {
    children?: any
    card?: boolean
    dotType?: string
}

export const LoadingComponent = ({ children, card, dotType }: ILoadingComponentProps) => {
    if (card) {
        return (
            <CenterPiece>
                <Card>
                    <CardBody>
                        <Loading dotType={dotType}>{children}</Loading>
                    </CardBody>
                </Card>
            </CenterPiece>
        )
    }

    return <Loading dotType={dotType}>{children}</Loading>
}

LoadingComponent.defaultProps = {
    card: true,
    dotType: 'dot-bricks'
}

export default LoadingComponent
