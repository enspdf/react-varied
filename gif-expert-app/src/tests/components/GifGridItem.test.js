import { shallow } from "enzyme";
import React from "react";

import { GifGridItem } from "../../components/GifGridItem";

describe("Test <GifGridItem/>", () => {
  const title = "Sample Title";
  const url = "http://localhost/image.png";

  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  it("Should show the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should have a paragraph with title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  it("Should have an image sent in props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  it("Should have an animate__fadeIn class", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
