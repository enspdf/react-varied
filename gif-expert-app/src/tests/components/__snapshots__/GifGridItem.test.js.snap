// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGridItem/> Should show the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Sample Title"
    src="http://localhost/image.png"
  />
  <p>
    Sample Title
  </p>
</div>
`;
